addons/l10n_in_ewaybill_irn/models/l10n_in_ewaybill.py

```diff
@@ -12,17 +12,21 @@ class L10nInEwaybill(models.Model):
 
     _inherit = 'l10n.in.ewaybill'
 
-    is_process_through_irn = fields.Boolean(compute='_compute_is_process_through_irn')
+    is_process_through_irn = fields.Boolean(compute='_compute_is_process_through_irn', store=True)
 
-    @api.depends('account_move_id')
+    @api.depends('account_move_id.l10n_in_edi_status')
     def _compute_is_process_through_irn(self):
         for ewaybill in self:
             account_move_id = ewaybill.account_move_id
+            if (
+                ewaybill.state in ('generated', 'cancel')
+                or not account_move_id
+                or account_move_id.move_type == 'out_refund'
```

Comment:
```suggestion
                or not account_move_id
                # E-way Bill is not generated by IRN for document types of Debit Note and Credit Note.
                or account_move_id.move_type == 'out_refund'
```
It should be move a longed with the comments